opencog/nlp/learn/scm/link-pipeline.scm

```diff
@@ -447,14 +460,19 @@
 "
  observe-text-public -- update word and word-pair counts by observing raw text.
  
- There are currently two observing modes, set by observe-mode, both taking
- an integer parameter:
+ There are currently three observing modes, set by observe-mode and taking
+ another integer parameter:
  - any: counts pairs of words linked by the LG parser in 'any' language.
- 	   'count-reach' specifies how many linkages from LG-parser to use.
+ 	    In this case, 'count-reach' specifies how many linkages from LG-parser
+ 	    to use.
  - clique: itearates over each word in the sentence and pairs it with
            every word located within distance 'count-reach' to its right.
            Distance is defined as the difference between words positions
            in the sentence, so neighboring words have distance of 1.
+ - clique-dist: same word-pairs as 'clique', but each word-pair is counted
+ 				a number of times determined by the distance between words
+ 				in the pair as:
+ 				(quotient count-reach distance)
```

Comment:
On a related note: do you really need to add this as a feature to the code?  Is this really needed?  I don't think it is.  The natural language world already knows that this si a stupid way to count: you will get bad results -- you will simply confirm what other people have measured decades ago, going back to the 1960's even.  It is well-known that word-pair observational distances don't work this way; a uniform-weight window of size six is needed for modern English, and for Shakespearian English, you need to move to a distance of at least twelve.  A modern review of the literature can be found in writings by Ferrar-i-Concho.  I don't have the reference on the tip of my fingers.

The problem is: when something is known to not work, we should not be adding it as a configurable parameter to the code.  If you want to experiment with this, you can certainly hand-edit the scheme, run the experiments, and see what happens.  But writing new code, passing a new parameter, turning the system into a spaghetti of pointless glorp just makes future code maintenance that much harder.  KISS == Keep it simple, sexy. Uness you really really need this parameter, and confirmed a real need for it with experiments, don't just add it because it sounds like a good idea.